### PR TITLE
Output memory config for conv2d and max_pool2d

### PR DESCRIPTION
**Pull request by mywoodstock**
### Ticket
#13106 

### Problem description
Need optional output memory config

### What's changed
`memory_config` parameter added (optional).

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/ttnn/unit_tests/operations/test_maxpool2d.py [hunk 1]
```diff
@@ -107,6 +107,7 @@ def run_max_pool(
         stride=[stride_h, stride_w],
         padding=[pad_h, pad_w],
         dilation=[dilation_h, dilation_w],
+        # memory_config=ttnn.DRAM_MEMORY_CONFIG,
```
Comment (bbradelTT) [inline tests/ttnn/unit_tests/operations/test_maxpool2d.py:110]: Please don't add commented out code.

If you want to test a specific memory config add code for it. If you want to say that a specific config will be used just add a comment stating that.

**Review by bbradelTT — APPROVED** [review]

**Review by cfjchu — APPROVED** [review]

**Review by pavlejosipovic — CHANGES_REQUESTED** [review]
Add at least some tests to make sure that optional mem config is applied

File: tests/ttnn/unit_tests/operations/test_new_conv2d.py [hunk 2]
```diff
@@ -797,6 +799,111 @@ def test_resnet50_conv_wh(
...
+@skip_for_blackhole()
+@pytest.mark.parametrize("device_params", [{"l1_small_size": 16384}], indirect=True)
+@pytest.mark.parametrize(
+    "batch_size, output_channels, input_channels, input_height, input_width, filter_height, filter_width, stride_h, stride_w, pad_h, pad_w, use_1d_systolic_array, config_override",
+    (
+        (16, 64, 16, 115, 115, 4, 4, 1, 1, 0, 0, True, {"act_block_h": 256}),
+        (8, 64, 64, 56, 56, 3, 3, 1, 1, 1, 1, True, None),
+    ),
+)
+@pytest.mark.parametrize(
```
Comment (pavlejosipovic) [inline tests/ttnn/unit_tests/operations/test_new_conv2d.py:812]: No need for params that have only one option, they just polute the file

File: tests/ttnn/unit_tests/operations/test_new_conv2d.py [hunk 3]
```diff
@@ -797,6 +799,111 @@ def test_resnet50_conv_wh(
...
+        pad_w,
+        use_1d_systolic_array,
+        config_override=config_override,
+        use_shallow_conv_variant=use_shallow_conv_variant,
+        transpose_mcast=use_1d_systolic_array,  ## use RM (transpose_mcast=False) with 2D on WH
+        packer_l1_acc=packer_l1_acc,
+        fp32_accum=False,
+        has_bias=has_bias,
+        auto_shard=auto_shard,
+        memory_config=memory_config,
```
Comment (pavlejosipovic) [inline tests/ttnn/unit_tests/operations/test_new_conv2d.py:903]: if we don't assert that this is the memory layout of the output tensor, there is no much point in having this test

**Review by pavlejosipovic — CHANGES_REQUESTED** [review]
> Add at least some tests to make sure that optional mem config is applied 

I still don't see any tests for maxpool?

**Review by pavlejosipovic — APPROVED** [review]

**Review by sankarmanoj-tt — APPROVED** [review]

**Pull request merged**